add caveat to check https://lsif.dev/

northyg opened the pull request:
## Test plan
Ensured link went to correct place, and was recently made aware that https://lsif.dev/ is the place to check for LSIF status.
This doc update was prompted by Zendesk ticket [#7154](https://sourcegraph.zendesk.com/agent/tickets/7154) in which I had the fun of re-remembering that page existed. I wish I had remembered this before filing an unnecessary bug report for lsif-py 🙃 The hope is that in the future when others reference this LSIF troubleshooting doc, they will avoid a similar fate.
<!--
  As part of SOC2/GN-104 and SOC2/GN-105 requirements, all pull requests are REQUIRED to
  provide a "test plan". A test plan is a loose explanation of what you have done or
  implemented to test this, or why this change does not need testing, as outlined in our
  Testing principles and guidelines:
  https://docs.sourcegraph.com/dev/background-information/testing_principles
  Write your test plan here after the "## Test plan" header.
-->

macraig commented:
@northyg I'd recommend checking our [own indexer docs](https://docs.sourcegraph.com/code_intelligence/references/indexers ) first as these are the ones we build inhouse and recommend using (we also own maintaining them if there's a bug).

However, if a customer references an indexer that's not listed in our docs, checking lsif.dev might help as there are indexers that we don't own listed there. In this case, the status might be relevant to flag to the customer along with who created and owns maintaining the indexer.

northyg commented:
> @northyg I'd recommend checking our [own indexer docs](https://docs.sourcegraph.com/code_intelligence/references/indexers) first as these are the ones we build inhouse and recommend using (we also own maintaining them if there's a bug).
> 
> However, if a customer references an indexer that's not listed in our docs, checking lsif.dev might help as there are indexers that we don't own listed there. In this case, the status might be relevant to flag to the customer along with who created and owns maintaining the indexer.

@macraig Thanks for the feedback; I'm glad I added you as a reviewer :) 
In light of this new info, I re-workerd my edit into the "When are issues related to code-intelligence?" section, and reverted the changes in the other. What do you think?

macraig reviewed (approved):
Thanks for documenting this!
